COMPILING.md

```diff
@@ -1,85 +1,319 @@
+
+E.g. for curses build on Debian and derivatives you'll also need `libncurses5-dev`.
+
+## Make flags
+
+Given you're building from source you have a number of choices to make:
+
+  * `NATIVE=` - you should use `linux64` if you're on a 64 bit system, `linux32` for 32 bit systems, and leave it out if you're building on Cygwin
+  * `RELEASE=1` - without this you'll get a debug build (see note below)
+  * `TILES=1` - if you want the tiles version; note that with this the ncurses version won't be build
+  * `SOUND=1` - if you want sound; this requires `TILES=1`
+  * `LOCALIZE=0` - this disables localizations so `gettext` is not needed
+  * `LUA=1` - this enables Lua *debug* support, may be useful if you plan writing mods
+  * `CLANG=1` - use Clang instead of GCC
+  * `CACHE=1` - use ccache
+
+There is a couple of other possible options - feel free to read the `Makefile`.
+
+If you have a multi-core computer you'd probably want to add `-jX` to the options, where `X` should roughly be twice the number of cores you have available.
+
+Example: `make -j4 CLANG=1 CCACHE=1 NATIVE=linux64 RELEASE=1 TILES=1`
+
+The above will build a tiles release for 64 bit Linux, using Clang and ccache and 4 parallel processes.
+
+**Note on debug**:
+You should probably always build with `RELEASE=1` unless you experience segfaults and are willing to provide stack traces.
+
+# Debian
+
+Instructions for compiling on a Debian-based system. The package names here are valid for Ubuntu 12.10 and may or may not work on your system.
+
+Building instructions, below, always assume you are running them from the Cataclysm:DDA source directory.
+
+## Linux (native) ncurses builds
 
-## Linux (native) ncurses builds ##
 Dependencies:
+
   * ncurses
   * build essentials
 
-```
-sudo apt-get install libncurses5-dev build-essential
-```
+    sudo apt-get install libncurses5-dev build-essential
+
+### Building
 
-### Building ###
-```
-make
-```
+    make
+
+## Cross-compiling to linux 32-bit from linux 64-bit
 
-## Cross-compiling to linux 32-bit from linux 64-bit ##
 Dependencies:
+
   * 32-bit toolchain
   * 32-bit ncurses
 
-```
-sudo apt-get install libc6-dev-i386 lib32stdc++-dev g++-multilib lib32ncurses5-dev
-```
+    sudo apt-get install libc6-dev-i386 lib32stdc++-dev g++-multilib lib32ncurses5-dev
+
+### Building
 
-### Building ###
-```
-make NATIVE=linux32
-```
+    make NATIVE=linux32
+
+## Cross-compile to Windows from Linux
 
-## Cross-compile to Windows from Linux ##
 Dependencies:
+
   * [mxe](http://mxe.cc)
 
-```
-sudo apt-get install autoconf bison flex cmake git automake intltool libtool scons yasm
-mkdir -p ~/src/mxe
-git clone -b stable https://github.com/mxe/mxe.git ~/src/mxe
-cd ~/src/mxe
-make gcc glib
-```
-
-### Building ###
-```
-PATH="${PATH}:~/src/mxe/usr/bin"
-make CROSS=i686-pc-mingw32-
-```
-
-## Linux (native) SDL builds ##
+    sudo apt-get install autoconf bison flex cmake git automake intltool libtool scons yasm
```

Comment:
Looks like you lost some backticks here -- this renders as HTML text, not preformatted.